account_invoice_report_payment_info/models/account_invoice.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2020 Tecnativa - Carlos Dauden
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.move"
+
+    def _get_reconciled_info_JSON_values(self):
+        res = super()._get_reconciled_info_JSON_values()
+        if not res:
+            return res
+        info_pattern = (
+            self.env["ir.config_parameter"]
+            .sudo()
+            .get_param("account_invoice_report_payment_info.info_pattern", default="")
+        )
+        Payment = self.env["account.move.line"]
+        for payment_dict in res:
+            payment = Payment.browse(payment_dict["payment_id"])
+            payment_dict["move_ref"] = payment.move_id.ref
+            payment_dict["extra_info"] = info_pattern.format(**payment_dict)
```

Comment:
What happen if extra_info is `False` 
IMO it's possible with a credit in customer invoice

```suggestion
            extra_info = info_pattern.format(**payment_dict)
            if extra_info: 
                payment_dict["extra_info"] = extra_info
```

Reply:
The extra_info deppends on the system parameter defined. With the option that you suggest and with the system parameter defined on data, the extra_info variable will take the value "(False)" so it will be evaluated as True. 

The problem of this, is that we don't know what will be defined on the system parameter, so it's not possible to avoid this problem.